Asterix/optics/propagation_functions.py

```diff
@@ -585,20 +586,24 @@ def prop_fpm_regional_sampling(pup,
 
     AUTHOR: R. Galicher (in IDL)
             ILa (to Python)
+            J. Mazoyer : small modifs to adapt to Asterix Dec 22
 
     Parameters
     ----------
-    pup : 2D array
+    pup : 2D array of size [self.dim_overpad_pupil,self.dim_overpad_pupil]
         Input mage array containing the wavefront at the entrance pupil of the optical system.
     fpm : 2D array
         Complex electric field in the focal plane of the focal-plane mask.
-    nbres : 1D array or list
+    real_dim_input : int or None, default None
+            Diameter of the support in pup (can differ from pup.shape). If None real_dim_input = pup.shape
+            Example: real_dim_input = diameter of the pupil in pixel for a padded pupil
+    nbres : list
         List of the number of resolution elements in the total image plane for all propagation layers.
+        As a general rule, it is probably safer to put these numbers so that there is not sampling shift right in
+        the middle of the DH to avoid confusing the matrix. So if we correct the DH between
+        two radius IWA and OWA (in lambda/D), nbrs should not have any elements between 2*IWA and 2*OWA.
```

Comment:
Could clarify the language:
"List of the number of resolution elements *across* the total image plane for all propagation layers.
As a general rule, it is probably safest to choose these numbers such that there is not sampling shift right in the middle of the DH. This could help in avoiding to confuse the matrix (which matrix?). So if we correct the DH between two radii IWA and OWA (in lambda/D), nbres should not have any elements between 2*IWA and 2*OWA."

Careful, I also fixed some typos here so that's minimum.

Reply:
done

Reply:
I just saw that the markdown made by `*` disappear in `2*IWA` and `2*OWA` xD If you feel like it, you could correct this, but it's low priority and I won't block the PR because of this.

Reply:
I'll do in the next PR directly. Because I have move this function to another file in the next PR, I need to do all these modification twice because they are not passed on :-D